Specify required-features for examples

Hello,

It is currently not possible to just run `cargo test` as this also builds the examples which require different features to be enabled.
Trying to run an example with e.g. `cargo run --example actix_webapi_oauth_interception_basic` also leads to error messages like

```
error[E0433]: failed to resolve: could not find `actix` in `zitadel`
  --> examples/actix_webapi_oauth_interception_basic.rs:2:14
   |
2  | use zitadel::actix::introspection::{IntrospectedUser, IntrospectionConfigBuilder};
   |              ^^^^^ could not find `actix` in `zitadel`
   |
note: found an item that was configured out
  --> /home/frederik/Documents/zitadel-rust/src/lib.rs:39:9
   |
39 | pub mod actix;
   |         ^^^^^
note: the item is gated behind the `actix` feature
  --> /home/frederik/Documents/zitadel-rust/src/lib.rs:38:7
   |
38 | #[cfg(feature = "actix")]
   |       ^^^^^^^^^^^^^^^^^

```

If the required features are specified in `Cargo.toml` a better error message is shown when trying to run an example for the above example we get:

```
error: target `actix_webapi_oauth_interception_basic` in package `zitadel` requires the features: `actix`
Consider enabling them by passing, e.g., `--features="actix"`
```

This also enable the usage of `cargo test` as it now knows that the examples are not usable without the features.
See https://doc.rust-lang.org/cargo/commands/cargo-test.html#target-selection where it says the following about which targets are build:

> bins (only if integration tests are built and required features are available)

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -177,3 +177,36 @@ http-body-util = "0.1.0"
 
 [package.metadata.docs.rs]
 all-features = true
+
+
+[[example]]
+name = "actix_webapi_oauth_interception_basic"
+required-features = ["actix"]
+
+[[example]]
+name = "axum_webapi_oauth_interception_basic"
+required-features = ["axum"]
+
+[[example]]
+name = "fetch_profile_with_pat"
+required-features = ["api", "interceptors"]
+
+[[example]]
+name = "fetch_profile_with_service_account"
+required-features = ["api", "interceptors"]
+
+[[example]]
+name = "rocket_webapi_oauth_interception_basic"
+required-features = ["rocket"]
+
+[[example]]
+name = "rocket_webapi_oauth_interception_jwtprofile_cached"
+required-features = ["rocket", "introspection_cache"]
+
+[[example]]
+name = "rocket_webapi_oauth_interception_jwtprofile"
+required-features = ["rocket"]
+
+[[example]]
+name = "service_account_authentication"
+required-features = ["credentials"]

